src/charm.py

```diff
@@ -58,8 +45,24 @@ def __init__(self, *args):
             "crds": "metacontroller-crds-v1.yaml",
             "rbac": "metacontroller-rbac.yaml",
             "controller": "metacontroller.yaml",
+            "service": "metacontroller-svc.yaml",
         }
 
+        # Observability integration
+        self.dashboard_provider = GrafanaDashboardProvider(self)
+        self.prometheus_provider = MetricsEndpointProvider(
+            charm=self,
+            relation_name="metrics-endpoint",
+            jobs=[
+                {
+                    "metrics_path": METRICS_PATH,
+                    "static_configs": [
+                        {"targets": [f"{self._name}-svc.{self._namespace}.svc:{METRICS_PORT}"]}
+                    ],
+                }
+            ],
+        )
+
```

Comment:
Also, I think we usually have the `observe` lines after everything in the `init()` function, right? At first, I thought this was the way but then I stumbled on this [seldon PR](https://github.com/canonical/seldon-core-operator/pull/236/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307c) the refactored the charm the opposite way.

Reply:
There is not a strong suggestion in the library's documentation, but I think that we could set a standard for all our charms. In general I see the obs charms (alertmanager, traefik, etc.) instantiating the `MetricsEdnpointProvider` before the `observe`.

Reply:
Yes exactly. But in this PR, the `observe` have been moved before the instantiation.

Reply:
I can update it to have it before the `observe` ^_^

Reply:
Updated, wdyt?

Reply:
Great!